### DISCUSSION TITLE
Write JSON from Lua object.

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#556] [Q&A] [answered]
Hi Richard,

Is there a way, during the Lua process, to generate out-of-the-box data ? 
Today I have a Lua process to generate tiles that contain all relations with the tag route=foot.
I'd like as well to generate another JSON (or anything similar) that would contain the `name`, `id` and `bbox`, directly in the same Lua process (or if necessary, in another run). For instance, once scanned, and used in `relation_function`, instead of only writing layers, would there be a way as well to write those features in another object that has basically nothing to do with TileMaker ? 

If so, is there a way to access the geometry of the object being written ? and then post-process it with `turf` for instance ?

**Comment by systemed** [comment, marked as answer]
Sorry, I missed this one initially. No, there isn't at present.

We have a few Lua instances running simultaneously, so I guess building this would require a common in-memory store, Lua wrappers to write to it, and a bunch of GeoJSON serialisation code. Probably quite complex for a fairly niche case.

You could perhaps have a go at writing something in your Lua code that opened a new file on `init_function`, appended to it as required in `way_function`/`node_function`, and closed it in `exit_function`. At present you'd be limited to writing the object ID and the object centroid - the Lua scripts don't have access to any other part of the geometry. Because there are several Lua instances, you'd probably want to generate a unique filename in `init_function` for each time it was called.

**Reply by BenOnTrack** [reply]
Thanks for the tips!
